### ISSUE TITLE
Cannot register a system

### ISSUE THREAD

**Issue by vastmeridian** [alphaess-developer/alphacloud_open_api#55]
Despite having an account and the serial number / check code of the inverter, I cannot register the system in alphacloud_open_api.

- The verification email takes too long to arrive (sometimes 45 seconds)
- The code window only stays open for 60seconds - despite the email saying the code is valid for 30minutes
- The site just says 'The Operation Failed' whenever I paste the code into the box and click save.

How do I resolve this?

**Comment by ElevenFan** [comment]
You can log in to the https://cloud.alphaess.com/ platform to change your email address, and then use your new email address to obtain the verification code.

**Comment by vastmeridian** [comment]
I have now been successful in registering my system and it is associated with my AppID.  However I am getting the following errpr when I attempt to set up my Home Assistant integration.

![image](https://github.com/alphaess-developer/alphacloud_open_api/assets/15128874/fff8cd68-2018-4d02-93c7-c675194ed163)

**Comment by CharlesGillanders** [comment]
Is home assistant showing any error in the log files?

**Comment by vastmeridian** [comment]
> Is home assistant showing any error in the log files?

Yes it does.
Unexpected json_response : {'code': 6007, 'msg': 'Sign check error'} when calling https://openapi.alphaess.com/api/getEssList
'NoneType' object is not iterable

There is a traceback error logged too.  Did you want that posted here or in your Issues log?

**Comment by Poshy163** [comment]
Hmm, going to copy this from another issue as it will give you a better idea:

[here](https://www.postman.com/poshy163/workspace/alphaess/collection/11219653-785a4a03-182a-46ea-991c-e841cd2da260?action=share&creator=11219653) is the "public" collection. just click on folder (called AlphaESS open API), then on variables and replace all the current value fields in appID, Appsecret and systemSN.

Then control + s to save and click on the "get ess list" on the side bar. and click on the blue send button?

What sort of response do you get?

**Comment by vastmeridian** [comment]
Did all that - thanks for the response.

Got this back
![image](https://github.com/alphaess-developer/alphacloud_open_api/assets/15128874/360f83e5-eaf4-4e47-a956-02b6218dfd50)

**Comment by vastmeridian** [comment]
Once I'd forked my own space, I had this response from AlphaESS

{"code":6007,"msg":"Sign check error"}

**Comment by vastmeridian** [comment]
And, when I log into AlphaCloud I have this:
![image](https://github.com/alphaess-developer/alphacloud_open_api/assets/15128874/43b3a6dd-a880-40e9-9fc7-0d5e36fff250)

**Comment by Poshy163** [comment]
Right, the error 6007 relates to a "Sign verification error". 

i only seem to get that error when i have a incorrect appID and app secret. are you sure the exact Appid and secret are making it onto the postman/in home assistant on setup?

**Comment by vastmeridian** [comment]
The two values are copied directly from Alpha's Cloud site using the copy alongside the values.  I am careful not to include any spaces before or after the values.

I think I may try regenerating the key, but I am happy to let you have the values privately to let you try directly.

I could maybe try resetting the AppSecretand trying again?

**Comment by Poshy163** [comment]
Sure, try resetting the app secret and give it a go, otherwise feel free to send it to me over twitter/X (same username and same profile pic) and ill give it a go.

its strange assuming you are using the right credentials...

**Comment by vastmeridian** [comment]
Resetting the AppSecret seems to have done the trick.

Thanks for your help.  I'll see if the integration completes in HA and I am able to read entity values and then close the issue.

**Comment by vastmeridian** [comment]
Entities now reading.  I'll close this as complete now.

**Comment by vastmeridian** [comment]
I tried once again (and with the same appID and appSecret) to use the alpha_api_demo.py script to access the openAPI and I **still** get the same 6007 sign check error response.  I've edited the alpha_api_demo.py script to include my appID and appSecret.

```
C:\Users\Test\Python\alphaess>python alpha_api_demo.py
Get ess list error: 6007 Sign check error
None

C:\Users\Test\Python\alphaess>
```
I get the same response from setting up and using Postman too.

```
{
    "code": 6007,
    "msg": "Sign check error"
}
```
Yet these same credentials are in use within the Home Assistant integration without a problem.  I'm loathe to reset appSecret again in case it breaks the Home Assistant integration or I lose my data.

Where am I going wrong?

**Comment by CharlesGillanders** [comment]
> 
> Where am I going wrong?
> 
> 
> 
>
> 

The signature algorithm includes a time stamp - I wonder if you have checked if your windows PC is locked to a valid NTP source so that it isn't running fast or slow?

**Comment by ElevenFan** [comment]
What are you doing? This doesn't look like our system.

**Comment by vastmeridian** [comment]
> > Where am I going wrong?
> 
> The signature algorithm includes a time stamp - I wonder if you have checked if your windows PC is locked to a valid NTP source so that it isn't running fast or slow?

I did wonder about that - but I get the same result from Postman.com, which would tend to rule that out.  Even so, there should be 300s leeway according to Alpha's documentation.

**Comment by vastmeridian** [comment]
> What are you doing? This doesn't look like our system.

@ElevenFan, it is Alpha's system and Alpha's demo code.  I tried the approach with @CharlesGillanders library first, then I tried using Postman, then I looked at this repository.  I was running your alpha_api_demo.py script.  The error comes from line 72 in this script.

What I am trying to do is to develop a script to get the statistics from my system from when it was installed.  At the moment I can't start this because I cannot get access to my system on your API because of this 6007 code.

I notice your appSecret in your example starts 7d71d896xxxxxxxx.  Mine starts a117e05cxxxxxxxx.  Does this matter?  

Would it upset the HA integration if I change my appSecret again - and why do I need to change this?
